### PR TITLE
Ruby Enumerable

### PR DESCRIPTION
**Pull request by Bhekisile**
### In this repository

- We implemented  the `MyList` class with the ` MyEnumerable` module inside the class and instance variable, `@list`
- The module includes methods such as `all?`, `any?` and `filter` 
- Added `README.md` file

### CODE REVIEW

**Review by Bantealem — COMMENTED** [review]
Hi @Bhekisile and @arnoldnekemiah , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has unprofessional readme. Currently, your readme doesn't have `Future Features` header and content. Please include `Future Feature` header and content to your project and ask for review.
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require_relative 'my_enumerable'
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (descholar-ceo) [inline my_list.rb:10]: - [ ] [OPTIONAL] You should handle a case where a user forgot to pass a block, you can add a statement like the following example:
```ruby
    return to_enum(:each) unless block_given?
```
- [ ]  [OPTIONAL]  You are not supposed to use the `each` method from ruby, you should implement your own each method programmatically, this project's requirements say this `MyList class implements #each method that yields all members of @list in succession.`, could you try to respect this requirement please?

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+end
+
+list = MyList.new(1, 2, 3, 4)
+list.each { |e| puts e }
+puts(list.all? { |e| e < 5 })
+puts(list.all? { |e| e > 5 })
+puts(list.any? { |e| e == 2 })
+puts(list.any? { |e| e == 5 })
+puts list.filter { |e| e.even? }.inspect # rubocop:disable Style/SymbolProc
```
Comment (descholar-ceo) [inline my_list.rb:19]: - [ ]  [OPTIONAL] You should not disable rubocop rules, instead look for a way to fix any linter issue related

**Review by descholar-ceo — APPROVED** [review]
Hi @Bhekisile && @arnoldnekemiah 🤝 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Your project works as expected, but I have left you rooms of improvement ✔️ 
👍🏽 PR follows the correct git flow 👍🏽 
👍🏽 Readme file has all sections ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (Bhekisile) [reply my_list.rb:10]: We will attend to it, thank you

File: my_list.rb (hunk 2 above)
Comment (Bhekisile) [reply my_list.rb:19]: We will fit it, thank you.

**Pull request merged**